extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java

```diff
@@ -0,0 +1,670 @@
+import software.amazon.awssdk.services.sagemaker.model.CreateEndpointResponse;
+import software.amazon.awssdk.services.sagemaker.model.CreateModelRequest;
+import software.amazon.awssdk.services.sagemaker.model.CreateModelResponse;
+import software.amazon.awssdk.services.sagemaker.model.DeleteEndpointConfigRequest;
+import software.amazon.awssdk.services.sagemaker.model.DeleteEndpointRequest;
+import software.amazon.awssdk.services.sagemaker.model.DeleteModelRequest;
+import software.amazon.awssdk.services.sagemaker.model.DescribeEndpointConfigRequest;
+import software.amazon.awssdk.services.sagemaker.model.DescribeEndpointRequest;
+import software.amazon.awssdk.services.sagemaker.model.DescribeModelRequest;
+import software.amazon.awssdk.services.sagemaker.model.ProductionVariant;
+import software.amazon.awssdk.services.sagemaker.model.SageMakerException;
+import software.amazon.awssdk.services.sagemaker.waiters.SageMakerWaiter;
+import software.amazon.awssdk.services.sagemakerruntime.SageMakerRuntimeClient;
+import software.amazon.awssdk.services.sagemakerruntime.model.InvokeEndpointRequest;
+import software.amazon.awssdk.services.sagemakerruntime.model.InvokeEndpointResponse;
+import software.amazon.awssdk.services.sts.StsClient;
+
+/** A utility class that help deploy model to SageMaker. */
+public final class SageMaker {
+
+    private static final Logger logger = LoggerFactory.getLogger(SageMaker.class);
+    private static final char[] CHARS = "abcdefghijklmnopqrstuvwxyz1234567890.-".toCharArray();
+
+    private SageMakerClient sageMaker;
+    private SageMakerRuntimeClient smRuntime;
+    private S3Client s3;
+    private IamClient iam;
+    private Region region;
+    private Model model;
+    private String modelName;
+    private String bucketName;
+    private String bucketPath;
+    private String executionRole;
+    private String containerImage;
+    private String endpointConfigName;
+    private String endpointName;
+    private String instanceType;
+    private int instanceCount;
+
+    private SageMaker(Builder builder) {
+        sageMaker = builder.sageMaker;
+        smRuntime = builder.smRuntime;
+        s3 = builder.s3;
+        iam = builder.iam;
+        model = builder.model;
+        modelName = model.getName();
+        bucketName = builder.bucketName;
+        bucketPath = builder.bucketPath;
+        executionRole = builder.executionRole;
+        containerImage = builder.containerImage;
+        endpointConfigName = builder.endpointConfigName;
+        endpointName = builder.endpointName;
+        instanceType = builder.instanceType;
+        instanceCount = builder.instanceCount;
+        region = DefaultAwsRegionProviderChain.builder().build().getRegion();
+    }
+
+    /**
+     * Deploys the model to Amazon SageMaker hosting service.
+     *
+     * @throws IOException if failed upload model to S3
+     */
+    public void deploy() throws IOException {
```

Comment:
Maybe we can set something like below:

```
deploy(SageMaker.END_POINT);
```
There could be SageMaker.END_POINT, SageMaker.LAMBDA, SageMaker.ECS...

Reply:
What is `SageMaker.LAMBDA`  or `SageMaker.ECS`?